source/process/documentation-UItext-guidelines.rst

```diff
@@ -0,0 +1,193 @@
+
+- Use complete sentences.
+- Explain what has happened or can happen, and what can go wrong as a consequence.
+- If the message contains a question, phrase it in such a manner so that it has a Yes/No answer (unless you have specific action buttons for the message).
+
+- Examples:
+
+  - *The Enterprise license expires in 2 days. If you do not renew it, some features will be disabled on licence expiry.*
+  - *If you claim this AD/LDAP account, you will no longer be able to log in with your email. Do you want to continue?*
+
+Error messages
+~~~~~~~~~~~~~~~
+
+An error message is for telling somebody that something went wrong. Errors prevent a user from doing a task or accessing a feature till the error is resolved.
+
+- Use complete sentences.
+- If what went wrong isn't obvious, explain in one sentence.
+- If a solution or workaround isn't obvious, suggest one.
+- Examples:
+
+  - *This message is too long. Shorten it to 120 characters.*
+  - *The passwords do not match.*
+
+
+Guidelines for UI elements
+---------------------------
+Use this table when writing the text for UI elements such as windows, dialog boxes, labels, and prompts.
+
+.. list-table::
+   :widths: 10 30 10 15 35
+   :header-rows: 1
+
+   * - Element
+     - Image 
+     - Capitalization
+     - Phrasing
+     - Examples     
+   * - Menu
+     - |menu.png|
+     - All words
+     - - Noun, noun phrase, or verb
+       - No punctuation
+       - Not more than three words
+     - - *Members*
+       - *Account Preferences*
+       - *Log Out*
+   * - Tooltip
+     - |tooltip.png|
+     - First word
+     - - Sentence fragment or sentence
+       - No punctuation
+       - Include articles (*a*, *an*, *the*)
+     - - *Start a Zoom meeting*
+       - *Flag for following up*
+       - *Remove from this list*
+   * - In-field text
+     - |field.png|  
+     - First word
+     - - Sentence fragment, sentence, or word
+       - No punctuation
+       - Include articles (*a*, *an*, *the*)
+     - - *Add a comment*
+       - *Search*
+   * - Action button
+     - |action.png|
+     - All words
+     - - Verb or verb phrase
+       - No articles (*a*, *an*, *the*)
+       - Exceptions: *OK*, *Yes*, *No*
+     - - *Add Comment*
+       - *Edit*
+   * - Label before a UI element
+     - |label_before.png|
+     - First word
+     - - Noun, verb, or sentence fragment
+       - End with a colon
+       - Include articles (*a*, *an*, *the*)
+     - - *Sign in with:*
+       - *Other words, separated by commas:*
+   * - Label after a UI element
+     - |label_after.png|
+     - First word
+     - - Noun, verb, or sentence fragment
+       - No punctuation
+     - - *Channels grouped by type*
+       - *Alphabetically*
+   * - Help text
+     - |help.png|
+     - First word
+     - Complete sentences, with punctuation
+     - - *You can add 20 more people.*
+       - *People are invited automatically to join the channel.*
+   * - Title
```

Comment:
Same general note as above, How much description is used for these, or do we rely on the nested buttons and subsequent help text to do more describing in hierarchical order. We just need to got to consistency

For example:
![image](https://user-images.githubusercontent.com/14333569/67972195-6e5c8200-fbcb-11e9-815d-3a2ef337a3a1.png)

Reply:
For better UX, I would've suggested moving the help text to _before_ the radio buttons. Are nested components re-used (used at more than one place on the interface)? If they are, we should probably keep the help text outside of the nesting, so that the buttons can make sense on their own?

Reply:
I think typical settings UX has buttons before help text, do you have examples of them the other way around? 

Settings are the only place that uses this nesting UI (Account, Team and Channel settings modals)

Reply:
Yes, typical UX does have buttons before help text but in those cases, generally, the help text is placed closest to the button it belongs to. Like this:
![image](https://user-images.githubusercontent.com/12574108/68733916-55f25b80-05fe-11ea-815d-96e2f1dbecc0.png)

I cannot remember seeing a nested UI like the Settings example, but here, I'd think the hierarchy itself provides sufficient context to a user, I would think. Only, as a user, I would've preferred the nested UI to be something more like this:
![image](https://user-images.githubusercontent.com/12574108/68734308-f1d09700-05ff-11ea-9cd0-26c0dd7baa78.png)

What do you think?